light-client/cli/src/bin/main.rs

```diff
@@ -39,67 +73,130 @@ fn main() {
+            // TODO should this use ThickClient and chain-id?
+            let ch = ChannelBuilder::default_channel_builder(env.clone())
+                .connect_to_uri(node_uri, &logger);
+
+            let client_api = ConsensusClientApiClient::new(ch.clone());
+            let config = client_api
+                .get_node_config(&Default::default())
+                .expect("get_node_config failed");
+
+            let blockchain_api = BlockchainApiClient::new(ch);
+            let last_block_info = blockchain_api
+                .get_last_block_info(&Default::default())
+                .expect("get_last_block_info failed");
+
+            (config, last_block_info)
+        })
+        .unzip();
+
+    let node_ids = node_configs
+        .iter()
+        .map(|node_config| HexKeyNodeID {
+            responder_id: ResponderId::from_str(node_config.get_peer_responder_id()).unwrap(),
+            public_key: node_config
+                .get_scp_message_signing_key()
+                .try_into()
+                .unwrap(),
+        })
+        .collect::<Vec<_>>();
+
+    let quorum_set = QuorumSet {
+        threshold: node_configs.len() as u32,
+        members: node_ids.into_iter().map(Into::into).collect(),
+    };
+
+    let trusted_validator_set = TrustedValidatorSetConfig { quorum_set };
+
+    let trusted_validator_set_start_block = last_block_infos
+        .iter()
+        .map(|last_block_info| last_block_info.index)
+        .max()
+        .unwrap_or_default();
+
+    let light_client_verifier = LightClientVerifierConfig {
+        trusted_validator_set,
+        trusted_validator_set_start_block,
+        historical_validator_sets: Default::default(),
+        known_valid_block_ids: Default::default(),
+    };
+
+    println!(
+        "{}",
+        serde_json::to_string_pretty(&light_client_verifier).unwrap()
+    );
+}
+
+fn cmd_fetch_archive_blocks(
+    tx_source_urls: Vec<String>,
+    block_index: u64,
+    out_file: PathBuf,
+    light_client_verifier_config_path: Option<PathBuf>,
+    logger: Logger,
+) {
+    let block_data = tx_source_urls
+        .into_iter()
+        .map(|url| {
+            log::info!(logger, "Fetching block data from {}", url);
+            let rts = ReqwestTransactionsFetcher::new(vec![url], logger.clone())
+                .expect("failed creating ReqwestTransactionsFetcher");
+            rts.get_block_data_by_index(block_index, None)
+                .expect("failed fetching block data")
+        })
+        .collect::<Vec<_>>();
+
+    if let Some(path) = light_client_verifier_config_path {
+        let json_data =
+            fs::read_to_string(path).expect("failed reading LightClientVerifierConfig file");
+        let light_client_verifier_config: LightClientVerifierConfig =
+            serde_json::from_str(&json_data).expect("failed parsing LightClientVerifierConfig");
+        let light_client_verifer = LightClientVerifier::from(light_client_verifier_config);
```

Comment:
:upside_down_face: nit
```suggestion
        let light_client_verifier = LightClientVerifier::from(light_client_verifier_config);
```